### ISSUE TITLE
[🐛 Bug]: driver.Manage().Cookies.AddCookie(); with Native aot(c#)

### ISSUE THREAD

**Issue by katanakatana123** [SeleniumHQ/selenium#13363]
### What happened?

when i use  driver.Manage().Cookies.AddCookie(); i got the error OpenQA.Selenium.WebDriverArgumentException: invalid argument: missing 'name', 
i think the problem comes from here https://github.com/SeleniumHQ/selenium/blob/b16d0d8885bbae2a9763bc529156c022c407cf26/dotnet/src/webdriver/WebDriver.cs#L567, when native aot cannot resolve json class

### How can we reproduce the issue?

```shell
driver.Manage().Cookies.AddCookie();
```

### Relevant log output

```shell
OpenQA.Selenium.WebDriverArgumentException: invalid argument: missing 'name',
```

### Operating System

window 10

### Selenium version

newest

### What are the browser(s) and version(s) where you see this issue?

120

### What are the browser driver(s) and version(s) where you see this issue?

120

**Comment by titusfortner** [comment]
You can't add a cookie that doesn't exist. You need at least a name and value. 
https://www.selenium.dev/documentation/webdriver/interactions/cookies/?tab=csharp#add-cookie

@nvborisenko shouldn't this fail for not sending a cookie instance to the method?

**Comment by nvborisenko** [comment]
The issue is reproduced, thanks for the report.

It is related to Native AOT.

```csharp
driver.Url = "https://google.com";

driver.Manage().Cookies.AddCookie(new Cookie("a", "b"));
```

```
Unhandled Exception: OpenQA.Selenium.WebDriverArgumentException: invalid argument: missing 'name'
  (Session info: chrome=120.0.6099.130)
   at OpenQA.Selenium.WebDriver.UnpackAndThrowOnError(Response, String) + 0x511
   at OpenQA.Selenium.WebDriver.Execute(String, Dictionary`2) + 0x65
   at OpenQA.Selenium.CookieJar.AddCookie(Cookie) + 0x6a
```

**Comment by nvborisenko** [comment]
Ah, we have logs! :)

```
00:54:05.780 DEBUG HttpCommandExecutor: Executing command: [0e18a5deeb9cd0f0b4b44625d29d74f2]: addCookie {"cookie":{}}
00:54:05.781 TRACE HttpCommandExecutor: >> Method: POST, RequestUri: 'http://localhost:57283/session/0e18a5deeb9cd0f0b4b44625d29d74f2/cookie', Version: 1.1, Content: System.Net.Http.ByteArrayContent, Headers:
{
  Accept: application/json; charset=utf-8
  Content-Type: application/json; charset=utf-8
}
00:54:05.784 TRACE HttpCommandExecutor: << StatusCode: 400, ReasonPhrase: 'Bad Request', Version: 1.1, Content: System.Net.Http.HttpConnectionResponseContent, Headers:
{
  Cache-Control: no-cache
  Content-Length: 903
  Content-Type: application/json; charset=utf-8
}
00:54:05.784 DEBUG HttpCommandExecutor: Response: ( InvalidArgument: System.Collections.Generic.Dictionary`2[System.String,System.Object])
```

Here we can see that AddCookie command serialized improperly.

**Comment by nvborisenko** [comment]
As a workaround you can mark your method, which invokes `AddCookie` method, with the following attribute:

```
[DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(Cookie))]
```

I see even more places in selenium code base where json serialization doesn't work properly with Native AOT.

Hoping #13097  will resolve it.

**Comment by NCLnclNCL** [comment]
> As a workaround you can mark your method, which invokes `AddCookie` method, with the following attribute:
> 
> ```
> [DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(Cookie))]
> ```
> 
> I see even more places in selenium code base where json serialization doesn't work properly with Native AOT.
> 
> Hoping #13097 will resolve it.

thank, it working,

**Comment by Chandru92S** [comment]
fhjisdhbfgjisa

**Comment by nvborisenko** [comment]
Removing it from 4.17 milestone. This is rare case, when we want to be friends with, but cannot do it in discoverable future. I would like to keep it as opened, hoping #13097 resolve it.

**Comment by nvborisenko** [comment]
Using `System.Text.Json` helped here.

```csharp
using var driver = new ChromeDriver();
driver.Url = "https://google.com";
driver.Manage().Cookies.AddCookie(new Cookie("a", "b"));
```

And it serializes `Cookie` object ccorrectly:
```
11:14:14.271 DEBUG HttpCommandExecutor: Executing command: [ae60dd5de4b05b0d71cc809e490d0f43]: addCookie {"cookie":{"name":"a","value":"b","secure":false,"httpOnly":false}}
11:14:14.271 TRACE HttpCommandExecutor: >> Method: POST, RequestUri: 'http://localhost:58107/session/ae60dd5de4b05b0d71cc809e490d0f43/cookie', Version: 1.1, Content: System.Net.Http.ByteArrayContent, Headers:
{
  Accept: application/json; charset=utf-8
  User-Agent: selenium/4.24.0
  User-Agent: (.net windows)
  Content-Type: application/json; charset=utf-8
  Content-Length: 67
}
{"cookie":{"name":"a","value":"b","secure":false,"httpOnly":false}}
11:14:14.277 TRACE HttpCommandExecutor: << StatusCode: 200, ReasonPhrase: 'OK', Version: 1.1, Content: System.Net.Http.HttpConnectionResponseContent, Headers:
{
  Cache-Control: no-cache
  Content-Length: 14
  Content-Type: application/json; charset=utf-8
}
```

Please upgrade to **v4.24**

> [!IMPORTANT]  
> You also should enable reflection based serialization, like in `csproj` file:
> ```xml
> <PublishAot>true</PublishAot>
> <JsonSerializerIsReflectionEnabledByDefault>true</JsonSerializerIsReflectionEnabledByDefault>
> ```
> https://learn.microsoft.com/en-us/dotnet/core/compatibility/serialization/8.0/publishtrimmed

**Comment by nvborisenko** [comment]
Hi @NCLnclNCL, are you able to help us to test the preview version of selenium package which seems to be AOT compatible (excluding CDP)? We are looking for everybody who is interested in it. @katanakatana123 and you?

**Comment by nvborisenko** [comment]
It should be fixed by https://github.com/SeleniumHQ/selenium/pull/14574

**Comment by nvborisenko** [comment]
Verified, will be available in v4.26
